presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1767,21 +1770,37 @@ else if (context.getDriverInstanceCount().isPresent()) {
             Optional<Integer> hashChannel = node.getPartitioningScheme().getHashColumn()
                     .map(symbol -> node.getOutputSymbols().indexOf(symbol));
 
-            LocalExchange localExchange = new LocalExchange(node.getPartitioningScheme().getPartitioning().getHandle(), driverInstanceCount, types, channels, hashChannel);
+            PartitioningHandle partitioningHandle = node.getPartitioningScheme().getPartitioning().getHandle();
```

Comment:
The addition of local exchanges can't happen in `LocalExecutionPlanner` anymore and instead must happen in `AddLocalExchanges`.